src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -365,6 +365,10 @@ GenTree* Lowering::LowerBinaryArithmetic(GenTreeOp* binOp)
             binOp->ChangeOper(GT_AND_NOT);
             BlockRange().Remove(notNode);
         }
+        else
+        {
+            ContainCheckAndChain(binOp);
+        }
```

Comment:
Seems like this runs for arm32 too which is probably not correct?
Though the arm32 testing does not fail, which is surprising to me, so maybe I am missing something.

Reply:
It's probably the `TARGET_ARM64` checks inside the functions.
I had left them in `lower.cpp` due to thinking this may get expanded to X64 later. But, that shouldn't be a concern of this PR. Will move everything.